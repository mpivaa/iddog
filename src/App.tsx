import * as React from 'react';
import { SignUp } from '@app/auth';
import { Feed } from '@app/feed';
import { getLocation, PrivRoute } from '@app/router';
import { State } from '@app/store';
import { Layout } from '@app/ui';
import { connect } from 'react-redux';
import { Redirect, Route, Switch } from 'react-router-dom';

export function App() {
  return (
    <Layout>
      <Switch>
        <Route path="/signup" component={SignUp} />
        <PrivRoute path="/feed" component={Feed} />
        <Redirect from="/" to="/feed" />
      </Switch>
    </Layout>
  );
}

const mapState = (state: State) => ({
  location: getLocation(state),
});

export default connect(mapState)(App as React.SFC);
