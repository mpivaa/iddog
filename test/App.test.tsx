import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { App } from '@app/App';
import Setup from '@test/setup';

test('App renders without errors', () => {
  const component = renderer.create(
    <Setup>
      <App />
    </Setup>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
