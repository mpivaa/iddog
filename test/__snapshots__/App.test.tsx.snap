// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without errors 1`] = `
<div
  className="sc-bxivhb oUyRg"
  size="large"
>
  <div
    className="sc-bxivhb dHxAwy"
    size="small"
  >
    <div
      className="sc-htpNat dSBIma"
      direction="column"
      fill={true}
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="sc-EHOje cYMhyC"
        >
          The IDDOG
        </div>
        <input
          className="sc-bdVaJa eKhXFQ"
          onChange={[Function]}
          placeholder="your email"
          type="email"
          value=""
        />
        <div
          className="sc-ifAKCX iIcXGD"
        />
        <button
          className="sc-bwzfXH chmtcU"
        >
          Entrar
        </button>
      </form>
    </div>
  </div>
</div>
`;
